pyproject.toml

```diff
@@ -0,0 +1,46 @@
+[build-system]
+requires = ["setuptools >= 61.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "python-terrier"
+description="PyTerrier"
+requires-python = ">=3.8"
+authors = [
+  {name = "Craig Macdonald", email = "craig.macdonald@glasgow.ac.uk"},
+]
+maintainers = [
+  {name = "Craig Macdonald", email = "craig.macdonald@glasgow.ac.uk"},
+]
+readme = "README.md"
+classifiers = [
+  "Topic :: Text Processing",
+  "Topic :: Text Processing :: Indexing",
+  "Programming Language :: Python :: 3",
+  "License :: OSI Approved :: Mozilla Public License 2.0 (MPL 2.0)",
+  "Operating System :: OS Independent",
+]
+dynamic = ["version", "dependencies"]
+
+[tool.setuptools.dynamic]
+version = {attr = "pyterrier.__version__"}
+dependencies = {file = ["requirements.txt"]}
+
+#package_data={'': ['LICENSE.txt', 'requirements.txt', 'requirements-test.txt']}
```

Comment:
are these included?

Reply:
do we need requirements.txt included?

Reply:
they don't need to be included

Reply:
you're the expert on this new format - can you make the changes?